### ISSUE TITLE
Enable accessibility is causing InkWell double click

### ISSUE THREAD

**Issue by bstatev-via** [flutter/flutter#154552]
### Steps to reproduce

After updating to Flutter 3.24.1 some of the elements in our app started getting a double click. So I create a minimized version of the problem. It can be reproduced in other cases too. But the simplest is a Text wrapper in a Container wrapper in InkWell. If accessibility is enabled at the start, the onTap method of InkWell is called 2 times instead of 1.

### Expected results

Clicking on the Click me button opens the dialog only 1 time.

### Actual results

Clicking on the Click me button opens the dialog 2 times.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const ExampleApp());
  SemanticsBinding.instance.ensureSemantics();
}

class ExampleApp extends StatelessWidget {
  const ExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: DoubleClickExample(),
    );
  }
}

class DoubleClickExample extends StatefulWidget {
  const DoubleClickExample({super.key});

  @override
  State<DoubleClickExample> createState() => _DoubleClickExampleState();
}

class _DoubleClickExampleState extends State<DoubleClickExample> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Double Click Example'),
      ),
      body: Column(
        children: [
          Center(
            child: InkWell(
              onTap: () {
                _dialogBuilder(context);
              },
              child: Container(
                height: 50,
                color: Colors.blue,
                child: const Text('Click me'),
              ),
            ),
          ),
        ],
      ),
    );
  }

  Future<void> _dialogBuilder(BuildContext context) {
    return showDialog<void>(
      context: context,
      builder: (BuildContext context) {
        return AlertDialog(
          title: const Text('Basic dialog title'),
          content: const Text(
            'A dialog is a type of modal window that\n'
            'appears in front of app content to\n'
            'provide critical information, or prompt\n'
            'for a decision to be made.',
          ),
          actions: <Widget>[
            TextButton(
              style: TextButton.styleFrom(
                textStyle: Theme.of(context).textTheme.labelLarge,
              ),
              child: const Text('Cancel'),
              onPressed: () {
                Navigator.of(context).pop();
              },
            ),
          ],
        );
      },
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.1, on macOS 14.6.1 23G93 darwin-arm64, locale en-BG)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.87.1)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
@bstatev-via Thanks for the report.
I verified this issue by running on iOS and voiceover ON using latest stable version. I observed that the `onTap` method is called only once as expected upon double tapping `click me`. See below:

```
2024-09-04 11:27:08.934414+0530 Runner[3437:199320] [AXRuntimeCommon] Unknown client: Runner
2024-09-04 11:27:13.943609+0530 Runner[3437:199594] flutter: I am tapped
2024-09-04 11:27:23.846541+0530 Runner[3437:199594] flutter: I am tapped
```

Above log shows two entries as I performed action twice. If only once, then it shows the callback only once as expected. 
Also, the dialog opens once as expected.

In which platform are you seeing the reported behavior ? Android or iOS ? A short video or log that shows twice call to `onTap` will be helpful.

Also, see if [this](https://github.com/flutter/flutter/issues/148230) is related to your case or not.
